packages/node-core/src/utils/sync-helper.spec.ts

```diff
@@ -0,0 +1,283 @@
+    const statement = generateCreateIndexStatement(
+      mockModel.options.indexes as any,
+      mockModel.schema.name,
+      mockModel.tableName
+    );
+    expect(statement).toStrictEqual([
+      `CREATE  INDEX "0xf48017d5c5d3d768" ON "test"."${mockModel.tableName}" USING gist ("from_id", "_block_range");`,
+      `CREATE  INDEX "0xb91efc8ed4021e6e" ON "test"."${mockModel.tableName}"  ("id");`,
+    ]);
+  });
+  it('Generate table statement no historical, no multi primary keys', () => {
+    jest.spyOn(mockModel, 'getAttributes').mockImplementationOnce(() => {
+      return {
+        id: {
+          type: 'text',
+          comment: 'id field is always required and must look like this',
+          allowNull: false,
+          primaryKey: true,
+          fieldName: 'id',
+          _modelAttribute: true,
+          field: 'id',
+        },
+      };
+    });
+    const statement = generateCreateTableStatement(mockModel, 'test');
+
+    // Correcting the expected statement to reflect proper SQL syntax
+    const expectedStatement = `
+        CREATE TABLE IF NOT EXISTS "test"."test-table" (
+      "id" text NOT NULL PRIMARY KEY
+    );
+  
+COMMENT ON COLUMN "test"."test-table"."id" IS 'id field is always required and must look like this';`.trim();
+
+    expect(statement.trim()).toBe(expectedStatement);
+  });
+  it('Reference statement', () => {
+    const attribute = {
+      type: 'text',
+      comment: 'The account that transfers are made from',
+      allowNull: false,
+      primaryKey: false,
+      fieldName: 'fromId',
+      _modelAttribute: true,
+      field: 'from_id',
+      references: {
+        model: {
+          tableName: 'accounts',
+          table: 'accounts',
+          name: 'Account',
+          schema: 't-sync-4',
+          delimiter: '.',
+        },
+        key: 'id',
+      },
+      onDelete: 'NO ACTION',
+      onUpdate: 'CASCADE',
+    } as ModelAttributeColumnOptions;
+
+    const statement = formatReferences(attribute, 'test');
+    expect(statement).toBe(`REFERENCES "test"."accounts" ("id") ON DELETE NO ACTION ON UPDATE CASCADE`);
+  });
+  it('Ensure correct foreignkey statement', () => {
+    jest.spyOn(mockModel, 'getAttributes').mockImplementationOnce(() => {
+      return {
+        transferIdId: {
+          type: 'text',
+          comment: undefined,
+          allowNull: false,
+          primaryKey: false,
+          field: 'transfer_id_id',
+          references: {
+            model: {
+              schema: 'test',
+              tableName: 'transfers',
+            },
+            key: 'id',
+          },
+          onDelete: 'NO ACTION',
+          onUpdate: 'CASCADE',
+        },
+      } as any;
+    });
+
+    const v = addForeignKeyStatement(mockModel);
+    expect(v[0]).toBe(
+      `ALTER TABLE "test"."test-table"
+      ADD FOREIGN KEY (transfer_id_id) 
+      REFERENCES "test"."transfers" (id) ON DELETE NO ACTION ON UPDATE CASCADE;`
```

Comment:
Id should be escaped
```suggestion
      REFERENCES "test"."transfers" ("id") ON DELETE NO ACTION ON UPDATE CASCADE;`
```